sigmastate/src/main/scala/sigmastate/types.scala

```diff
@@ -1688,9 +1688,6 @@ object SCollection extends STypeCompanion with MethodByNameUnapply {
     */
   def flatMap_eval[A, B](mc: MethodCall, xs: Coll[A], f: A => Coll[B])
                         (implicit E: ErgoTreeEvaluator): Coll[B] = {
-    if (!VersionContext.current.isJitActivated) {
-      checkValidFlatmap(mc)
```

Comment:
Why check for pre-JIT interpretation removed? How this change is non-breaking?

Reply:
This code (flatMap_eval) belong to new v5.0 interpreter, it is not used in v4.x.
Will be used after v5.0 activation.

In the previous version of v5.0 I implemented the same constraints on flatMap as in v4.x.
The intention was to make ergoTree.version based versioning.
But, as in the other previous cases, tree based versioning was replaced with activation-based.
And, as the next step, the limitations where removed, because they don't make sense with JITC.